Sage/carbon
# fix(menu): variant type mismatch MenuItem and Link

### Proposed behaviour
Renamed the `variant` prop to `menuItemVariant` inside of the styled component in order to avoid an error when rendering the `MenuItem` as a `Link`.

![no-error](https://github.com/user-attachments/assets/7f52745c-710c-4a9e-b891-fa44dac74ce2)




### Current behaviour
When rendering the `MenuItem` as a `Link` an error is thrown due to the `variant` prop 
![error](https://github.com/user-attachments/assets/72426414-5c65-4ca8-ac37-7ece4a3571ea)



### Checklist




#### QA


### Additional context



### Testing instructions

Please check the comments in this task https://jira.sage.com/browse/FE-6700 for detailed testing instructions.
